### DISCUSSION TITLE
lmdeploy有generator api可以只返回output ids而不是output string吗？

### DISCUSSION THREAD

**Discussion by haoxufd** [InternLM/lmdeploy#1567] [Q&A]
想集成lmdeploy做推理，用于rlhf过程，这个过程需要得到一次generate的output ids，但是调用lmdeploy的pipeline，只能得到decode后的output str，请问有什么推荐的解决办法吗？
vllm有generate接口，可以一次处理一个batch的prompts，直接得到output_ids，lmdeploy有没有类似的接口呢？

**Comment by lvhan028** [comment]
Since v0.4.0, the pipeline can output logprobs and token ids. 
Unfortunately, we found a bug in that version and will fix it in PR https://github.com/InternLM/lmdeploy/pull/1561
